src/main/java/com/puppycrawl/tools/checkstyle/checks/UncommentedMainCheck.java

```diff
@@ -94,9 +94,6 @@ public void beginTree(DetailAST rootAST) {
     @Override
     public void leaveToken(DetailAST ast) {
         if (ast.getType() == TokenTypes.CLASS_DEF) {
-            if (classDepth == 1) {
```

Comment:
please remove filed `classDepth` completely from Class and use `currentClass` in all cases instead. 
`currentClass == null` === `classDepth = 0`
`currentClass != null` === `classDepth == 1`

Reply:
@romani Wouldn't this break the current behavior if there was more than 1 top level class?
We couldn't reset `currentClass` back to `null` without knowing the AST of the top level.

Reply:
`visitToken` should initialize `currentClass` and `leaveToken` should reset it to null.
It should work, but it is better to recheck .

Reply:
ok, could be done in separate issue - #5064